astropy/io/fits/convenience.py

```diff
@@ -897,7 +897,7 @@ def tabledump(filename, datafile=None, cdfile=None, hfile=None, ext=1,
         if not datafile:
             # TODO: Really need to provide a better way to access the name of
             # any files underlying an HDU
-            root, tail = os.path.splitext(f._HDUList__file.name)
+            root, tail = os.path.splitext(f._file.name)
```

Comment:
Could you remove the TODO above, as it was precisely about this use of `_HDUList__file`.